### DISCUSSION TITLE
SI - Smith-Nightmare portrait?

### DISCUSSION THREAD

**Discussion by r-h-o-n-a** [exult/exult#311] [Show and tell]
Hi all - just wondering if anyone's managed to fix the issue with the implementation of Smith's portrait in the Dream Realm in SI?

For funsies, while I was poking around in Exult Studio a little while ago I thought I'd do up a Smith-Nightmare portrait, for if I'd be able to work out how to implement it. If anyone would like to make use of the portrait itself, here it is - opinions always appreciated.

![faces vga s320_f0](https://user-images.githubusercontent.com/93699238/213588902-f7d150cd-afbc-4200-abd7-66cafba950a8.png)

**Comment by DominusExult** [comment]
This references #154 

I dabbled with it but couldn'T get it to work. I wonder if I could get it to work with an actual face to show.
The face looks nice :)

**Comment by KnightCaptainU7** [comment]
The nightmare horse shape code is shape#(0x375), and you could set the FACE_SMITH there.

**Reply by r-h-o-n-a** [reply]
Awesome. Is this doable in Exult Studio? I'm afraid I'm still stumped on how to use Exult Tools to get the usecode decompiled into something I can productively work with. (After trying every permutation of ucxt command line syntax that I can think of, as well as reinstalling both SI and the newest version of Exult Tools to make sure the paths line up (following advice from old forum posts at sourceforge), I still can't get ucxt to produce any rational output. And for some reason ucxt is the only app in the Exult Tools package that doesn't have a readme included, so I'm pretty much flailing here.)

I emailed you at the end of December about this, though I also realise it was probably a time of year when emails would easily slip through :) but I could really use a hand with decompiling the usecode into a state I can productively work with. Can you (or anyone else with the expertise) help?

**Reply by DominusExult** [reply]
Yes, you can assign the face in Exult Studio (start Exult, enter mapedit code, in Exult Studio's shape browser edit the nightmare shape (and first add this face to faces.vga as a new shape so you can select it in the nightmare shape edit)).
I don't know if our nightmare code prevents this from working.

Ucxt documentation is at https://github.com/exult/exult/tree/master/usecode/ucxt/Docs

I'll have to see why it's not added to the tools snapshot. Can't do anything until next week because backpain prevents me from sitting down 🤣

**Reply by r-h-o-n-a** [reply]
Awesome, thank you. (I'm massively keen to throw myself deep into the modding as well as the translation work - the early steps of learning how to get the original code into a workable shape are just really opaque at first, even for someone who grew up in an MS-DOS environment.) It's past 2 in the morning here so I'll leave doing anything with it till tomorrow but will update if I have any joy with it. :)

Also, for sure, your back should be #1 priority! 😲 I blew two discs myself in 2007 and have been dealing with the aftermath ever since, so I know how rough it can get. Take it easy on yourself!

**Reply by DominusExult** [reply]
I've added the UCXT documentation to the tools installer. Next snapshot of tools will have it.

**Comment by DominusExult** [comment]
I can't fix that it doesn't show a face, nor am I able to add a custom face.
I'm adding the shape of your png to the bug report (I had to convert it to shp)

**Comment by KnightCaptainU7** [comment]
![image](https://user-images.githubusercontent.com/117867844/225473329-86aa1a34-77f1-4bdd-b4af-f6948891811c.png)

**Comment by KnightCaptainU7** [comment]
I made a pull request!
https://github.com/exult/exult/pull/322

Might be helpful to you, @r-h-o-n-a, for seeing an example of the code changes needed to support this sort of change.

**Reply by KnightCaptainU7** [reply]
@marzojr Thank you for editing and merging the pull. I also appreciate the feedback on various lines.
